jdaviz/app.py

```diff
@@ -1071,24 +1090,48 @@ def _get_range_subset_bounds(self, subset_state,
             return [{"name": subset_state.__class__.__name__,
                      "glue_state": subset_state.__class__.__name__,
                      "region": spec_region,
+                     "sky_region": None,  # not implemented for spectral, but include for consistancy
                      "subset_state": subset_state}]
         return spec_region
 
     def _get_multi_mask_subset_definition(self, subset_state):
+
         return [{"name": subset_state.__class__.__name__,
                  "glue_state": subset_state.__class__.__name__,
                  "region": subset_state.total_masked_first_data(),
+                 "sky_region": None,
                  "subset_state": subset_state}]
 
-    def _get_roi_subset_definition(self, subset_state):
-        # TODO: Imviz: Return sky region if link type is WCS.
+    def _get_roi_subset_definition(self, subset_state, to_sky=False):
+
+        sky_subset_region = None
+        override_wcs = None  # only for cubeviz, where gwcs is missing spatial info
+        if to_sky:
+            wcs = subset_state.xatt.parent.coords
+            if self.config == 'cubeviz':
+                parent_data = subset_state.attributes[0].parent
+                print('PARENT DATA META', parent_data.meta)
+                override_wcs = parent_data.meta["_orig_spec"].wcs
+
+        # pixel region
         roi_as_region = roi_subset_state_to_region(subset_state)
+
+        # sky region
+        roi_as_sky_region = None
+        if to_sky:
+            roi_as_sky_region = roi_subset_state_to_region(subset_state,
+                                                           to_sky=True,
+                                                           override_wcs=override_wcs)
+
         return [{"name": subset_state.roi.__class__.__name__,
                  "glue_state": subset_state.__class__.__name__,
                  "region": roi_as_region,
+                 "sky_region": roi_as_sky_region,
                  "subset_state": subset_state}]
 
-    def get_sub_regions(self, subset_state, simplify_spectral=True, use_display_units=False):
+    def get_sub_regions(self, subset_state, simplify_spectral=True,
+                        use_display_units=False, get_sub_regions=False,
```

Comment:
I don't think `get_sub_regions` is used anywhere?
```suggestion
                        use_display_units=False,
```

Reply:
its used for composite subsets

Reply:
Right, but `get_sub_regions` is a method and you are using it here as an argument.

Reply:
oh whattt, i have no idea when/why i added that. removed.